display_information/sketch-on-map/src/test/java/com/esri/samples/sketch_on_map/SketchOnMapTest.java

```diff
@@ -0,0 +1,351 @@
+  }
+
+  /**
+   * Test Case 5: Use the SketchEditor to draw a freehand polygon on the map, and verify that it is added to the GraphicsOverlay.
+   */
+  @Test
+  public void drawFreehandPolygon() {
+    clickOn(freehandPolygonButton);
+    clickAndDragFourPoints();
+    clickOn(saveButton);
+
+    ListenableList<Graphic> graphics = graphicsOverlay.getGraphics();
+    assertEquals(1, graphics.size());
+    assertEquals(GeometryType.POLYGON, graphics.get(0).getGeometry().getGeometryType());
+  }
+
+  /**
+   * Clicks on four different points in the map view in succession, starting from the center.
+   */
+  private void clickOnFourPoints() {
+    clickOn(mapView);
+    moveBy(-100, 0);
+    clickOn(MouseButton.PRIMARY);
+    moveBy(0, 100);
+    clickOn(MouseButton.PRIMARY);
+    moveBy(100, 0);
+    clickOn(MouseButton.PRIMARY);
+  }
+
+  /**
+   * Clicks and drags the mouse to four different points, starting from the center of the map view.
+   */
+  private void clickAndDragFourPoints() {
+    drag(mapView)
+        .moveBy(-150, 0)
+        .moveBy(0, 150)
+        .moveBy(150, 0)
+        .drop();
+  }
+
+  /**
+   * Test Case 6: Create a point and edit it.
+   */
+  @Test
+  public void editPoint() {
+    // create a point sketch
+    clickOn(pointButton);
+    clickOn(mapView);
+    clickOn(saveButton);
+
+    // click on the point and edit it
+    Graphic point = graphicsOverlay.getGraphics().get(0);
+    Point2D originalLocation = mapView.locationToScreen((Point) point.getGeometry());
+    moveTo(mapView, Pos.TOP_LEFT, originalLocation, Motion.DIRECT);
+    clickOn(MouseButton.PRIMARY);
+    clickOn(editButton);
+
+    moveTo(mapView, Pos.TOP_LEFT, originalLocation, Motion.DIRECT);
+    moveBy(100, -100);
+    clickOn(MouseButton.PRIMARY);
+    clickOn(saveButton);
+
+    Point2D newLocation = mapView.locationToScreen((Point) point.getGeometry());
+    assertNotEquals(originalLocation, newLocation);
+  }
+
+  /**
+   * Test Case 7: Create a polyline and edit it by moving a vertex
+   */
+  @Test
+  public void editPolylineMoveVertex() {
+    // create a polyline sketch
+    clickOn(polylineButton);
+    clickOnFourPoints();
+    clickOn(saveButton);
+
+    // move to the bottom right corner of the polygon and select it
+    Graphic graphic = graphicsOverlay.getGraphics().get(0);
+    Geometry oldGeometry = graphic.getGeometry();
+    double xMin = graphic.getGeometry().getExtent().getXMin();
+    double yMin = graphic.getGeometry().getExtent().getYMin();
+    Point2D originaBottomLeftCorner = mapView.locationToScreen(new Point(xMin, yMin, mapView.getSpatialReference()));
+    moveTo(mapView, Pos.TOP_LEFT, originaBottomLeftCorner, Motion.DIRECT);
+    clickOn(MouseButton.PRIMARY);
+
+    // start editing
+    clickOn(editButton);
+
+    // move to the bottom left vertex, select it, and move it
+    moveTo(mapView, Pos.TOP_LEFT, originaBottomLeftCorner, Motion.DIRECT);
```

Comment:
Seem to be a few different ways of dragging going on

Reply:
The way it was just there, it's actually adding an additional vertex, not moving an existing one. I hadn't realise it's click and drag to move a vertex, not click and click.
Updated to actually drag the existing vertex somewhere else.